beamer/deploy/commands.py

```diff
@@ -0,0 +1,90 @@
+import subprocess
+from pathlib import Path
+from typing import Any
+
+import click
+import structlog
+
+import beamer.contracts
+import beamer.deploy.artifacts
+import beamer.deploy.config
+import beamer.util
+from beamer.deploy.util import deploy_contract, get_commit_id
+from beamer.typing import ChainId
+from beamer.util import make_web3
+
+log = structlog.get_logger(__name__)
+
+
+def _ensure_commit_is_on_remote() -> None:
+    commit_id = get_commit_id()
+    output = subprocess.check_output(["git", "branch", "-r", "--contains", commit_id])
+    branches = map(str.strip, output.decode("utf-8").split("\n"))
+    if not any(
+        branch == "origin/main" or branch.startswith("origin/release/") for branch in branches
+    ):
+        raise RuntimeError("could not find commit %s on the remote" % commit_id)
+
+
+class _ChainIdParam(click.ParamType):
+    def convert(self, value: Any, param: click.Parameter | None, ctx: click.Context | None) -> Any:
+        try:
+            chain_id = ChainId(int(value))
+            if chain_id <= 0:
+                raise ValueError("chain ID must be positive")
+        except ValueError as exc:
+            self.fail(str(exc), param, ctx)
+        else:
+            return chain_id
+
+
+@click.command("deploy-base")
+@click.option(
+    "--rpc-file",
+    type=click.Path(exists=True, file_okay=True, dir_okay=False, path_type=Path),
+    required=True,
+    metavar="PATH",
+    help="Path to the RPC config file.",
+)
+@click.option(
+    "--keystore-file",
+    type=click.Path(exists=True, file_okay=True, dir_okay=False, path_type=Path),
+    required=True,
+    metavar="PATH",
+    help="Path to the keystore file.",
+)
+@click.password_option(
+    "--password",
+    type=str,
+    default="",
+    prompt=False,
+    help="The password needed to unlock the keystore file.",
+)
+@click.option(
+    "--artifacts-dir",
```

Comment:
general question, did we ensure the change in wording happened in the whole codebase? 

`output-dir` -> `artifacts-dir`
`deployment files` -> `deployment artifacts`

Reply:
We will need to update at least the agent, but I decided to leave all of that for later, we're still using the old deployment script. But yes, we need to update the relevant places to use new names.